org.subsurface_divelog.Subsurface.json

```diff
@@ -0,0 +1,123 @@
+{
+	"app-id": "org.subsurface_divelog.Subsurface",
+	"runtime": "org.kde.Platform",
+	"runtime-version": "5.9",
+	"sdk": "org.kde.Sdk",
+	"command": "subsurface",
+	"rename-icon": "subsurface-icon",
+	"rename-appdata-file": "subsurface.appdata.xml",
+	"rename-desktop-file": "subsurface.desktop",
+	"finish-args":[
+		"--socket=wayland",
+		"--socket=x11",
+		"--share=network",
+		"--device=dri",
+		"--filesystem=home",
+		"--device=all",
+		"--env=QTWEBENGINE_DISABLE_SANDBOX=1"
+	],
+	"build-options": {
+		"cflags": "-O2",
+		"cxxflags": "-O2"
+	},
+	"cleanup": [
+		"/include",
+		"/lib/pkgconfig",
+		"/lib64/pkgconfig",
+		"/lib/debug",
+		"/lib/cmake",
+		"/share/doc",
+		"/share/man",
+		"*.a",
+		"*.la",
+		"*.cmake"
+	],
+	"modules":[
+		"shared-modules/udev/udev-175.json",
+		{
+			"name":"libssh2",
+			"buildsystem":"cmake",
+			"sources":[
+				{
+					"type":"git",
+					"url":"https://github.com/libssh2/libssh2.git"
+				}
+			]
+		},
+		{
+			"name":"grantlee",
+			"buildsystem":"cmake",
+			"sources":[
+				{
+					"type":"git",
+					"url":"https://github.com/steveire/grantlee.git"
+				}
+			]
+		},
+		{
+			"name":"libzip",
+			"no-autogen":true,
+			"sources":[
+				{
+					"type":"git",
+					"url":"https://github.com/nih-at/libzip.git"
+				},
+				{
+					"type":"shell",
+					"commands":["autoreconf -vfi", "./configure --prefix=/app", "make"]
```

Comment:
Creating a file named `autogen.sh` with `script` would be better. You could also ditch `no-autogen` then.